e2e/ansible/playbooks/resiliency/test-k8s-master-failure/k8s-master-failure.yml

```diff
@@ -0,0 +1,311 @@
+
+
+     - name: 6)  Create and Build one sample job (Hello World)  inside pod.
+       shell: >
+         source ~/.profile; kubectl exec {{ jenkins_pod_name.stdout }} -n {{ namespace }} 
+         -- bash -c "java -jar /var/jenkins_home/jenkins-cli.jar -s
+         http://localhost:8080 create-job sample < /var/jenkins_home/template.xml";
+         source ~/.profile; kubectl exec {{ jenkins_pod_name.stdout }} -n {{ namespace  }} 
+         -- bash -c "java -jar /var/jenkins_home/jenkins-cli.jar -s http://localhost:8080 build sample"
+       args:
+         executable: /bin/bash
+       delegate_to: "{{groups['kubernetes-kubemasters'].0}}"
+
+
+     - name: 6a)  Running a build job. 
+       shell: >
+          source ~/.profile; kubectl exec {{ jenkins_pod_name.stdout }} -n {{ namespace }}
+          -- bash -c "java -jar /var/jenkins_home/jenkins-cli.jar -s http://localhost:8080 console sample"
+       args:
+         executable: /bin/bash
+       delegate_to: "{{groups['kubernetes-kubemasters'].0}}"
+       register: First_Build
+
+
+     - name: Bring down the kubemaster.
+       include_tasks: "{{ansible_env.HOME}}/{{utils_path}}/vagrant-halt-up-task.yml"
+       vars:
+         vagrant_op: halt --force
+         nodename: kubemaster-01
+
+
+     - name: wait for 120 seconds.
+       wait_for:
+          timeout: 120
+
+
+
+     - name: 7) Checking the status of kubemaster after kubemaseter halt.
+       shell: vagrant status kubemaster-01 | grep kubemaster-01 |  awk '{ $1="";print}' | cut -d '(' -f1
+       args:
+         chdir: "{{ansible_env.HOME}}/{{vagrant_path}}"
+         executable: /bin/bash
+       register: VDown
+       until: "'poweroff'  in VDown.stdout"
+
+
+     - name: Bring up the kubemaster.
+       include_tasks: "{{ansible_env.HOME}}/{{utils_path}}/vagrant-halt-up-task.yml"
+       vars:
+         vagrant_op: up 
+         nodename: kubemaster-01
+
+     - name: wait for 120 seconds.
+       wait_for:
+          timeout: 120
+
+     - name: 7a) Checking the status of kubemaster.
+       shell: vagrant status kubemaster-01 | grep kubemaster-01 |  awk '{ $1="";print}' | cut -d '(' -f1
+       args:
+         chdir: "{{ansible_env.HOME}}/{{vagrant_path}}"
+         executable: /bin/bash
+       register: Vup
+       until: "' running '  in Vup.stdout"
+
+
+     - name:  Confirm pod status is running after kubemaster up.
+       include_tasks: "{{ansible_env.HOME}}/{{utils_path}}/deploy_check.yml"
+       vars:
+          ns: "{{ namespace }}"
+          app: jenkins
+
+      
+ 
+     - name: 8) Confirm volume container is running
+       shell: source ~/.profile; kubectl get pods -n {{ namespace }} | grep pvc | grep {{item}} | grep Running | wc -l
+       args:
+        executable: /bin/bash
+       register: result
+       until: result.stdout|int >= 1
+       delay: 30
+       retries: 20
+       with_items:
+        - ctrl
+        - rep
+       delegate_to: "{{ groups['kubernetes-kubemasters'].0 }}"
+
```

Comment:
This condition will pass even if only one replica is running.

Reply:
Yes, I have changed my task which actually looks for all replica's.